더보기 화면 작업

sunshiningsoo opened the pull request:
WIP!!
기본 틀이랑 동작만 작업해서 우선 올리려 합니다.
미리 리뷰로 조언 많이 받습니다,, ㅎ

![Simulator Screen Recording - iPhone 15 Pro - 2023-11-30 at 11 53 14](https://github.com/ku-ring/ios-app/assets/72736657/0a2ab1a7-3cc5-451c-9302-f9275b4d7508)

sunshiningsoo committed:
[추가] 더보기 기본 메뉴 구성 및 네비게이션, 모달 액션

sunshiningsoo committed:
[추가] 프로퍼티 바인딩

sunshiningsoo committed:
[삭제] 카카오링크 삭제

sunshiningsoo committed:
[수정] BindingState 로 변경

x-0o0 commented:
진행중이라 PR 에서 Draft 로 변경했습니다~ 작업 완료되시면 Ready for review 클릭해주시면 됩니다.

<img width="959" alt="스크린샷 2023-11-30 오후 5 31 09" src="https://github.com/ku-ring/ios-app/assets/53814741/3a8827c4-f59f-4f48-88b0-7880b0b87d4c">

x-0o0 commented on KuringApp/KuringApp/Setting/FeedbackView.swift:
```diff
@@ -0,0 +1,114 @@
+                    .cornerRadius(8)
+                    .background(
+                        RoundedRectangle(cornerRadius: 12)
+                            .stroke(.green, lineWidth: 1)
```
쿠링색(초록색) 사용하는 부분은 `Color.accentColor` 를 사용하도록 바꿔주세요

x-0o0 commented on KuringApp/KuringApp/Setting/FeedbackView.swift:
```diff
@@ -0,0 +1,114 @@
+                    .multilineTextAlignment(.center)
+                
+                VStack {
+                    TextEditor(text: viewStore.binding(
+                        get: \.text,
+                        send: { .typing($0) })
+                    )
```
`viewStore.binding(get:send:)` 말고 `State/text` 에 `@BindingState` 속성을 추가해주는게 어떨까요?

x-0o0 replied:
https://github.com/x-0o0/SwiftUI-TCA-Note#binding

여기에 간략하게 정리해뒀습니다!

x-0o0 commented on KuringApp/KuringApp/Setting/FeedbackView.swift:
```diff
@@ -0,0 +1,114 @@
+        Reduce { state, action in
+            switch action {
+            case .eraseView:
+                return .run { send in
+                    await send(.delegate(.erase))
+                }
```
delegate 용도면 `.run` 말고 바로 `send` 사용하시면 됩니다.

<img width="811" alt="스크린샷 2023-11-30 오후 6 03 54" src="https://github.com/ku-ring/ios-app/assets/53814741/3c3d0b5c-6060-4b7f-a30b-911aa9b17766">

x-0o0 commented on KuringApp/KuringApp/Setting/FeedbackView.swift:
```diff
@@ -0,0 +1,114 @@
+    enum Action: Equatable {
+        case eraseView
+        
+        case delegate(Delegate)
+        enum Delegate: Equatable {
+            case erase
+        }
```
`PresentationState` 를 `nil` 시키는 용도라면 delegate 로 직접 `nil` 처리 할 필요가 없습니다. TCA 가 dismiss 될때 알아서 해줍니다.

x-0o0 commented on KuringApp/KuringApp/Setting/OpenSourceView.swift:
```diff
@@ -0,0 +1,76 @@
+    
+    var body: some View {
+        WithViewStore(self.store, observe: { $0 }) { viewStore in
+            List(Opensource.items, id: \.self) { item in
```
item 이 `Identifiable`를 준수하면 id 파라미터에 값을 할당하지 않아도 됩니다.
```suggestion
            List(Opensource.items) { item in
```

x-0o0 commented on KuringApp/KuringApp/Setting/OpenSourceView.swift:
```diff
@@ -0,0 +1,76 @@
+extension Opensource {
+    static let items: [Opensource] = [
+        Opensource(name: "TCA - The Composable Architecture", link: "https://github.com/pointfreeco/swift-composable-architecture/tree/main"),
+        Opensource(name: "Lottie", link: "https://github.com/airbnb/lottie-ios"),
+        Opensource(name: "KuringCommons", link: "https://github.com/KU-Stacks/kuring-ios-commons"),
```
제거 부탁드립니다~ 안쓸거라서요!

x-0o0 commented on KuringApp/KuringApp/Setting/SettingView.swift:
```diff
@@ -8,10 +8,23 @@
         var currentAppIcon: KuringIcon?
+        @BindingState var isCustomAlarmOn: Bool = false
+        
+        @PresentationState var webViewOpen: InformationWebViewFeature.State?
+        @PresentationState var feedbackViewPresent: FeedbackFeature.State?
```
```suggestion
        @PresentationState var informationWebView: InformationWebViewFeature.State?
        @PresentationState var feddback: FeedbackFeature.State?
```

Feature 앞에 해당하는 네이밍으로 쓰는게 어떨까요?

x-0o0 commented on KuringApp/KuringApp/Setting/SettingView.swift:
```diff
@@ -20,15 +33,72 @@ struct SettingListFeature: Reducer {
+        
+        // url 이동
+        case urlButtonTapped(String)
+        case eraseWebView(PresentationAction<InformationWebViewFeature.Action>)
+        
+        // 피드백 작성
+        case feedbackViewPresent
+        case eraseFeedbackView(PresentationAction<FeedbackFeature.Action>)
```
```suggestion
        case informationWebView(PresentationAction<InformationWebViewFeature.Action>)
        
        // 피드백 작성
        case feedbackViewPresent
        case feedback(PresentationAction<FeedbackFeature.Action>)
```

x-0o0 replied:
각 Feature의 PresentationAction 에 대한 것이기 때문에 액션을 `erase` 에만 제한을 두지 않는 네이밍이 좋아보입니다.

x-0o0 commented on KuringApp/KuringApp/Setting/SettingView.swift:
```diff
@@ -20,15 +33,72 @@ struct SettingListFeature: Reducer {
+                }
+                
+            case let .sns(url):
+                UIApplication.shared.open(URL(string: url)!)
+                return .none
```
```suggestion
                guard let url = URL(string: url) else { return .none }
                UIApplication.shared.open(url)
                return .none
```

x-0o0 committed:
[추가] KuringPackage 추가
- App 라이브러리 추가
- NoticeUI, SubscriptionUI, DepartmentUI, SearchUI 추가
- swift-composable-architecture 브랜치룰을 observation-beta 로 설정
- iOS 17 이상

x-0o0 committed:
[추가] ColorSet 모듈 추가
- UI 모듈에 디펜던시 추가

x-0o0 committed:
[수정] KuringPackage 적용 및 워닝 수정
- [수정] A single sheet is supported 관련 에러 및 Canvas 프리뷰 동작하지 않는 에러 해결 (동일 원인)
- [삭제] KuringPackage 적용하면서 불필요해진 앱프로젝트 소스코드 제거

x-0o0 committed:
[삭제] 옛날 패키지 제거

x-0o0 committed:
[추가] 패키지 테스트 타겟 추가 및 스크립트 추가

x-0o0 committed:
[수정] yaml syntax 오류 수정

x-0o0 committed:
[수정] 테스트 에러 수정

x-0o0 committed:
[추가] `-skipMacroValidation` 옵션 추가

x-0o0 committed:
[수정] 올바른 시뮬레이터 버전으로 테스트 하도록 수정

x-0o0 committed:
[수정] 테스트 개선

x-0o0 committed:
[수정] 트리거 커맨드 단순화

x-0o0 commented:
`feature/jaesung/more` 브랜치에서 작업 진행함에 따라 PR close 합니다
